scalalib/src/mill/javalib/android/AndroidSdkModule.scala

```diff
@@ -59,10 +66,22 @@ trait AndroidSdkModule extends Module {
    *
    * For More Read Bundle Tool [[https://developer.android.com/tools/bundletool Documentation]]
    */
-  def bundleToolPath: T[PathRef] = Task {
-    val bundleToolJar = Task.dest / "bundleTool.jar"
-    os.write(bundleToolJar, requests.get(bundleToolUrl()).bytes)
-    PathRef(bundleToolJar)
+  def bundleToolPath: T[PathRef] = Task() {
+    val url = bundleToolUrl()
+    // TODO: Use caching API once available, https://github.com/com-lihaoyi/mill/issues/3930
+    val cache = FileCache()
+      .pipe { cache =>
+        if (Task.offline) cache.withCachePolicies(Seq(LocalOnly)) else cache
+      }
+    cache.file(Artifact(url)).run.unsafeRun()(cache.ec) match {
+      case Right(file) =>
+        Result.Success(PathRef(os.Path(file)).withRevalidateOnce)
+      case Left(ex) if Task.offline =>
+        Result.Failure(s"Can't fetch bundle tools (from ${url}) while in offline mode.")
```

Comment:
This is me trying to get a feeling for the current coursier cache API. I'd like to work on that API next. Ideally, we can built-in offline suport.